scaffolding-go/lib/scaffolding.sh

```diff
@@ -29,23 +29,57 @@ scaffolding_load() {
 #
 
 # Path constuctor for GOPATH
-scaffolding_go_gopath=${scaffolding_go_gopath:-"$HAB_CACHE_SRC_PATH/$pkg_dirname"}
-export scaffolding_go_gopath
+export scaffolding_go_gopath=${scaffolding_go_gopath:-"$HAB_CACHE_SRC_PATH/$pkg_dirname"}
```

Comment:
This can be updated to use `SRC_PATH` which was added to plan-build after the initial go scaffolding was written.
https://github.com/habitat-sh/habitat/blob/master/components/plan-build/bin/hab-plan-build.sh#L3077-L3084

Eg: `export scaffolding_go_gopath=${scaffolding_go_gopath:-"$SRC_PATH"}`